### PR TITLE
interpreting index name in min_itemsize specification #11364, #10381

### LINKED ISSUE #10381: BUG: DataFrame.to_hdf doesn't pass along min_itemsize for index
Unless I'm seeing something wrong

``` python
In [21]: df = DataFrame(dict(A = 'foo', B = 'bar'),index=range(5)).set_index("A")

In [22]: df.to_hdf('store.h5', 'test', format='table', min_itemsize={'index': 10})

In [23]: store = pd.HDFStore('store.h5')

In [24]: store.get_storer('test').table
Out[24]:
/test/table (Table(5,)) ''
  description := {
  "index": StringCol(itemsize=3, shape=(), dflt=b'', pos=0),
  "values_block_0": StringCol(itemsize=3, shape=(1,), dflt=b'', pos=1)}   # <---- I think this should be 10
  byteorder := 'irrelevant'
  chunkshape := (10922,)
  autoindex := True
  colindexes := {
    "index": Index(6, medium, shuffle, zlib(1)).is_csi=False}
```

and FYI this raises (not sure if it should work)

``` python
In [25]: df.index.name = 'theindex'

In [26]: df.to_hdf('store.h5', 'test2', format='table', min_itemsize={'theindex': 10})
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
```

Just a report right now... no time.

### LINKED ISSUE #11364: inconisitent behaviour of hdf5 store.append for MultiIndex vs simple Index
The following example works with MultiIndex, but fails with simple index, which is bad in terms of generalization / intuitive behaviour:
```
#Create example:
col_nums = [0]
df = pd.DataFrame({"V1":["a","b","c","d","e", "aaaah!!!"], 
                          "W":["c","d","c","d","c","c"],
                          "data":np.arange(6)})
df.set_index(["V1","W"], inplace = True)
df.to_csv("testtable.tab",sep = "\t")
```

```
# MulitIndex -> works:
sep = "\t"
indexcols =[0,1]
chunksize=5

xbed = "testtable.tab"
%rm 'tempstore.h5'
# create a store
with pd.HDFStore('tempstore.h5') as store:
    for nn, chunk in enumerate(pd.read_table(xbed, chunksize=chunksize, sep = sep, index_col= indexcols)):
        group = "x"
        print(chunk.index.names)
        store.append(group, chunk, format = "table", 
                     min_itemsize=dict(zip(chunk.index.names, [32]*len(chunk.index.names))))
        print("chunk #" , nn, file = sys.stderr)
```

```
# simple Index -> fails:
sep = "\t"
indexcols =[0]    # <==== this is the only difference
chunksize=5

xbed = "testtable.tab"
%rm 'tempstore.h5'
# create a store
with pd.HDFStore('tempstore.h5') as store:
    for nn, chunk in enumerate(pd.read_table(xbed, chunksize=chunksize, sep = sep, index_col= indexcols)):
        group = "x"
        print(chunk.index.names)
        store.append(group, chunk, format = "table", 
                     min_itemsize=dict(zip(chunk.index.names, [32]*len(chunk.index.names))))
        print("chunk #" , nn, file = sys.stderr)
```
` -> ...  ValueError: min_itemsize has the key [V1] which is not an axis or data_column`

### PR DESCRIPTION
**Pull request by DSLituiev**
closes #10381 

The fix replaces named index columns in [`Table.index_axes` with their names](https://github.com/pydata/pandas/compare/master...DSLituiev:master#diff-1b15d1477da3a0548d2dd72a5d023d00L3323)  and [corrects `min_itemsize` if it contains 'index' item which is not in `Table.index_axes`](https://github.com/pydata/pandas/compare/master...DSLituiev:master#diff-1b15d1477da3a0548d2dd72a5d023d00R3093)

### CODE REVIEW

File: scripts/test_hdf5_index_11364.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+import os
+
+def create_test_file():
+    col_nums = [0]
+    df = pd.DataFrame({"V1":["a","b","c","d","e", "aaaah!!!"],
```
Comment (jreback) [inline scripts/test_hdf5_index_11364.py]: you need to craft a test along the lines of the tests in ``pandas/io/tests/test_pytables.py``

**Comment by jreback** [comment]
this is where to make the change: https://github.com/pydata/pandas/blob/master/pandas/io/pytables.py#L3248

**Comment by DSLituiev** [comment]
I updated the test, but sorry I do not see how to handle it by changing https://github.com/pydata/pandas/blob/master/pandas/io/pytables.py#L3248 as it works  on `data_columns` and the issue concerns the index columns.

**Comment by jreback** [comment]
tests should be actually in test_pytables.py

and just slot into the existing infrastructure
you are reinventing the wheel

**Comment by DSLituiev** [comment]
I am sorry to admit I am unfamiliar with your infrastructure and conventions. I am not a professional software developer and it is first time I am trying it. Does it need to come in a special place? Anyway, I do not get your idea with implementation.

**Comment by jreback** [comment]
guidlines are [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html)

no worries. just trying to help / get people to contribute.

**Comment by jreback** [comment]
closing, but if you'd like to update, pls reopen

**Comment by DSLituiev** [comment]
No problem. I anyway did not get how you want it to be solved. 
As I wrote  https://github.com/pydata/pandas/blob/master/pandas/io/pytables.py#L3248  works on data_columns and the issue concerns the index columns.

**Comment by jreback** [comment]
I think the references get moved. (as the file was updated). no worries. I was looking for a general fix ,rather than special casing here.

**Pull request closed without merging**